dartagnan/src/main/java/com/dat3m/dartagnan/analysis/Refinement.java

```diff
@@ -244,125 +224,74 @@ private static void logSummary(List<SolverStatistics> statList, int iterationCou
-            perms.add(Function.identity());
-
-            for (EquivalenceClass<Thread> c : symmClasses) {
-                List<Thread> threads = new ArrayList<>(c);
-                threads.sort(Comparator.comparingInt(Thread::getId));
+        return message;
+    }
 
-                switch (learningOption) {
-                    case NONE:
-                        break;
-                    case LINEAR:
-                        for (int i = 0; i < threads.size(); i++) {
-                            int j = (i + 1) % threads.size();
-                            perms.add(symm.createTransposition(threads.get(i), threads.get(j)));
+    // This code is pure debugging code that will generate graphical representations
+    // of each refinement iteration.
+    // Generate .dot files and .png files per iteration
+    private static void generateGraphvizFiles(RefinementTask task, ExecutionModel model, int iterationCount, DNF<CoreLiteral> reasons) {
+        //   =============== Visualization code ==================
+        // The edgeFilter filters those co/rf that belong to some violation reason
+        BiPredicate<EventData, EventData> edgeFilter = (e1, e2) -> {
+            for (Conjunction<CoreLiteral> cube : reasons.getCubes()) {
+                for (CoreLiteral lit : cube.getLiterals()) {
+                    if (lit instanceof RelLiteral) {
+                        RelLiteral edgeLit = (RelLiteral) lit;
+                        if (model.getData(edgeLit.getData().getFirst()).get() == e1 &&
+                                model.getData(edgeLit.getData().getSecond()).get() == e2) {
+                            return true;
                         }
-                        break;
-                    case QUADRATIC:
-                        for (int i = 0; i < threads.size(); i++) {
-                            for (int j = i + 1; j < threads.size(); j++) {
-                                perms.add(symm.createTransposition(threads.get(i), threads.get(j)));
-                            }
-                        }
-                        break;
-                    case FULL:
-                        List<Function<Event, Event>> allPerms = symm.createAllPermutations(c);
-                        allPerms.remove(Function.identity()); // We avoid adding multiple identities
-                        perms.addAll(allPerms);
-                        break;
-                    default:
-                        throw new UnsupportedOperationException("Symmetry learning option: "
-                                + learningOption.name() + " is not recognized.");
+                    }
                 }
             }
+            return false;
+        };
+
+        String programName = task.getProgram().getName();
+        programName = programName.substring(0, programName.lastIndexOf("."));
+        String directoryName = String.format("%s/output/refinement/%s-%s-debug/", System.getenv("DAT3M_HOME"), programName, task.getTarget());
```

Comment:
`output/refinement` is not in this branch. What happens if such folder does not exists? is it created?

Reply:
It will get created. The method that gets called uses `mkdirs` to generate all the necessary directories.